### PR TITLE
Upgrading samples to flutter_lints, part 1 of n

### PR DESCRIPTION
**Pull request by domesticmouse**
Breaking up the migration into a set of PRs

### CODE REVIEW

**Comment by domesticmouse** [comment]
FYI @mit-mit

File: animations/analysis_options.yaml [hunk 1]
```diff
@@ -7,25 +7,19 @@ analyzer:
...
-    - test_types_in_equals
-    - throw_in_finally
-    - unnecessary_brace_in_string_interps
-    - unnecessary_getters_setters
-    - unnecessary_new
-    - unnecessary_statements
+    avoid_types_on_closure_parameters: true
+    avoid_void_async: true
+    await_only_futures: true
+    camel_case_types: true
```
Comment (mit-mit) [inline animations/analysis_options.yaml:13]: Several of these seem to already be in the lint sets, so can't remove them from here?

Comment (domesticmouse) [reply animations/analysis_options.yaml:13]: Which lint set did I miss? I've trimmed some from our previous list of lints

Comment (mit-mit) [reply animations/analysis_options.yaml:13]: I just spot checked. But for example `camel_case_types` is in https://github.com/dart-lang/lints/blob/main/lib/core.yaml

Comment (mit-mit) [reply animations/analysis_options.yaml:13]: By lints sets, I meant is already in one of these three files (which are chained together by imports):
https://github.com/dart-lang/lints/blob/main/lib/core.yaml
https://github.com/dart-lang/lints/blob/main/lib/recommended.yaml
https://github.com/flutter/packages/blob/master/packages/flutter_lints/lib/flutter.yaml

Comment (domesticmouse) [reply animations/analysis_options.yaml:13]: Looks like `hash_and_equals` and `non_constant_identifier_names` are also now duplicates.

File: jsonexample/lib/json_strings.dart [hunk 2]
```diff
@@ -3,22 +3,22 @@
 // found in the LICENSE file.
 
 abstract class JsonStrings {
-  static final String listOfInts = '[1, 2, 3]';
+  static const String listOfInts = '[1, 2, 3]';
```
Comment (RedBrogdon) [inline jsonexample/lib/json_strings.dart:6]: nit: Could probably remove the `String` type annotations on these. This is some of the first Flutter code I ever wrote, which shows sometimes. 😄

**Review by RedBrogdon — APPROVED** [review]
LGTM. Words cannot express how tired my mousewheel finger is.

File: animations/lib/src/misc/animated_positioned.dart [hunk 3]
```diff
@@ -38,20 +38,20 @@ class _AnimatedPositionedDemoState extends State<AnimatedPositionedDemo> {
...
   Widget build(BuildContext context) {
     final size = MediaQuery.of(context).size;
-    final appBar = AppBar(title: Text('AnimatedPositioned'));
+    final appBar = AppBar(title: const Text('AnimatedPositioned'));
     final topPadding = MediaQuery.of(context).padding.top;
     // AnimatedPositioned animates changes to a widget's position within a Stack
     return Scaffold(
       appBar: appBar,
-      body: Container(
+      body: SizedBox(
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply animations/lib/src/misc/animated_positioned.dart:46]: Oddly enough, I wrote that lint about a year ago, after getting poked about it in code reviews one too many times =)

File: animations/analysis_options.yaml [hunk 4]
```diff
@@ -7,25 +7,15 @@ analyzer:
...
+    cancel_subscriptions: true
+    close_sinks: true
+    directives_ordering: true
+    file_names: false
+    package_api_docs: true
+    package_prefixed_library_names: true
+    test_types_in_equals: true
+    throw_in_finally: true
+    unnecessary_statements: true
+    use_key_in_widget_constructors: false
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply animations/analysis_options.yaml:21]: From memory, `file_names` is only required on one sample, so I'll strip it where not required.

On the other hand, `use_key_in_widget_constructors` is going to require adding constructors in a lot of places to pass through the optional `Key`. I wasn't sure how that'd look.

[the remark this answers is not in the document]
Comment (domesticmouse) [reply animations/analysis_options.yaml:21]: Raised https://github.com/flutter/samples/issues/807 to track this

File: animations/lib/src/misc/curved_animation.dart [hunk 5]
```diff
@@ -26,18 +26,18 @@ class _CurvedAnimationDemoState extends State<CurvedAnimationDemo>
...
-    CurveChoice(curve: Curves.easeOutCubic, name: 'Ease Out Cubic'),
-    CurveChoice(curve: Curves.easeInExpo, name: 'Ease In Expo'),
-    CurveChoice(curve: Curves.easeOutExpo, name: 'Ease Out Expo'),
-    CurveChoice(curve: Curves.elasticIn, name: 'Elastic In'),
-    CurveChoice(curve: Curves.elasticOut, name: 'Elastic Out'),
-    CurveChoice(curve: Curves.easeInQuart, name: 'Ease In Quart'),
-    CurveChoice(curve: Curves.easeOutQuart, name: 'Ease Out Quart'),
-    CurveChoice(curve: Curves.easeInCirc, name: 'Ease In Circle'),
-    CurveChoice(curve: Curves.easeOutCirc, name: 'Ease Out Circle'),
+    const CurveChoice(curve: Curves.bounceIn, name: 'Bounce In'),
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply animations/lib/src/misc/curved_animation.dart]: done

File: jsonexample/lib/json_strings.dart (hunk 2 above)
Comment (domesticmouse) [reply jsonexample/lib/json_strings.dart]: done

**Pull request merged**